osu.Game/Screens/OnlinePlay/Multiplayer/Participants/StateDisplay.cs

```diff
@@ -1,126 +1,169 @@
             {
                 AutoSizeAxes = Axes.Both,
+                Anchor = Anchor.CentreRight,
+                Origin = Anchor.CentreRight,
                 Spacing = new Vector2(5),
                 Children = new Drawable[]
                 {
-                    text = new OsuSpriteText
-                    {
-                        Anchor = Anchor.CentreLeft,
-                        Origin = Anchor.CentreLeft,
-                        Font = OsuFont.GetFont(weight: FontWeight.Regular, size: 12),
-                        Colour = Color4Extensions.FromHex("#DDFFFF")
-                    },
                     icon = new SpriteIcon
                     {
-                        Anchor = Anchor.CentreLeft,
-                        Origin = Anchor.CentreLeft,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.CentreRight,
                         Icon = FontAwesome.Solid.CheckCircle,
                         Size = new Vector2(12),
-                    }
+                    },
+                    new CircularContainer
+                    {
+                        Masking = true,
+                        AutoSizeAxes = Axes.Both,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.CentreRight,
+                        Children = new Drawable[]
+                        {
+                            progressBar = new ProgressBar(false)
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                BackgroundColour = Color4.Black.Opacity(0.4f),
+                                FillColour = colours.Blue,
+                                Alpha = 0f,
+                            },
+                            text = new OsuSpriteText
+                            {
+                                Padding = new MarginPadding { Horizontal = 5f, Vertical = 1f },
+                                Anchor = Anchor.CentreRight,
+                                Origin = Anchor.CentreRight,
+                                Font = OsuFont.GetFont(weight: FontWeight.Regular, size: 12),
+                                Colour = Color4Extensions.FromHex("#DDFFFF")
+                            },
+                        }
+                    },
                 }
             };
         }
 
-        protected override void LoadComplete()
-        {
-            base.LoadComplete();
-            updateStatus();
-        }
+        private OsuColour colours;
 
-        [Resolved]
-        private OsuColour colours { get; set; }
-
-        private void updateStatus()
+        public void UpdateStatus(MultiplayerUserState state, BeatmapAvailability availability)
         {
-            switch (status)
+            if (availability.State != DownloadState.LocallyAvailable)
```

Comment:
this is super back to front. it means that the download state overrides a potential game state. i think the download state should only show if the game state is idle, no?